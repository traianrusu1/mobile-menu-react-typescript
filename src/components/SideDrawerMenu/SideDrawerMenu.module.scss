@import "../../appStyles/variables";
@import "../MobileMenu/mobileMenuVariables";
.sideDrawerMenu {
  display: flex;
  flex-direction: column;
}

.childMenu {
  background-color: $appColor2;
}

.childMenuDynamic {
  background-color: red;
  position: relative;
  left: -200px;
}

.open {
  animation: slide-in 0.3s ease 0s 1 normal forwards;
}

.closed {
  animation: slide-out 0.3s ease 0s 1 normal forwards;
}

@keyframes slide-in {
  0% {
    left: -200px;
  }
  100% {
    left: 0;
  }
}
@keyframes slide-out {
  0% {
    left: 0px;
  }
  100% {
    left: -200px;
  }
}
