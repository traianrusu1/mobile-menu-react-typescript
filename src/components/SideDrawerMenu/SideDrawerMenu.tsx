import React from 'react';
import styles from './SideDrawerMenu.module.scss';
import { SideDrawerMenuItem as SideDrawerMenuItemInterface } from '../../interfaces/SideDrawerMenuItem';
import { SideDrawerMenuItem } from '..';

interface Props {
  menuItems: SideDrawerMenuItemInterface[];
}

const SideDrawerMenu: React.FC<Props> = ({ menuItems }: Props) => {
  return (
    <div className={styles.sideDrawerMenu}>
      {menuItems.map((item) => (
        <SideDrawerMenuItem key={item.name} item={item} />
      ))}
    </div>
  );
};

export default SideDrawerMenu;
