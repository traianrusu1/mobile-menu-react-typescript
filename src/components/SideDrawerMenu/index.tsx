import SideDrawerMenu from './SideDrawerMenu';

export default SideDrawerMenu;
