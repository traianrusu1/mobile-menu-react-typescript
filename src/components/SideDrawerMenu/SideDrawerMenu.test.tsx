import React from 'react';
import { render, cleanup } from '@testing-library/react';

import SideDrawerMenu from './SideDrawerMenu';

afterEach(cleanup);

describe('<SideDrawerMenu />', () => {
  test('renders', () => {
    const { container } = render(<SideDrawerMenu myProp="" />);
    expect(container).toMatchSnapshot();
  });
});

