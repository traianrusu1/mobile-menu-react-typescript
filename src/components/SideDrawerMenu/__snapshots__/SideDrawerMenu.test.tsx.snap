// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawerMenu /> renders 1`] = `
<div>
  <div
    class="sideDrawerMenu"
  >
    SideDrawerMenu
    
  </div>
</div>
`;
